### ISSUE TITLE
Session list behavior for impersonated sessions

### ISSUE THREAD

**Issue by PVince81** [owncloud/impersonate#76]
### Steps

1. As "admin", create a user "user1"
1. Open a session for "admin" on two browsers B1 and B2
1. In B1, impersonate user "user1"
1. In B2 where still as admin, go to settings and check the session list
1. Now still in B2, login as "user1" and check the session list

### Expected result
Session list of "admin" only shows one session, the one from the current window.
Session list of "user1" shows two sessions, the one from the current window and the one from the impersonating admin.

### Actual result
Session list of "admin" shows two sessions: the one from the current window and the one that was used for impersonating.
Session list of "user1" only shows the session from the current window.

### Versions
ownCloud 10.0.3
impersonate 0.1.0

Note that in the future core will have a feature for admins to kill sessions from any users (https://github.com/owncloud/core/issues/29548), so I'd expect that whatever is displayed for individual users would also be visible for admins.

The token manager stores the sessions in oc_authtoken and in the current behavior, it still has the one for "admin" even after impersonation. This is what is causing the issue from https://github.com/owncloud/impersonate/issues/74.

Depending on the resolution of #74, we could decide to create a tracking token after impersonating the user to see the behavior from "Expected result".

But we might also want to discuss whether a user should be able to see impersonation sessions from admins and even be able to kill these. Note that if we refrain from keeping track of these sessions internally and technically, no one would be able to kill said sessions in case of security breaches. So in general it is still a good idea to track all sessions as per "Expected behavior" above, but we might want to add a flag to tell whether users are supposed to see some of these entries (impersonate) in their list or not.

@sharidas @pmaier1 @Peter-Prochaska

**Comment by sharidas** [comment]
@PVince81 when I try to debug with phpstorm, I see that $_SESSION array has one entry for both the browsers. Hope I am not doing stupid by looking into $_SESSION variable. As per the new change we made, by invalidating the token and then recreating session token, this is expected right?

**Comment by PVince81** [comment]
Yeah I think it's fine for now. We do need to decide what we do expect eventually.
